frontend/public/components/namespace-dropdown.tsx

```diff
@@ -0,0 +1,309 @@
+    ? `${userSettingsPrefix}.systemNamespaceInSelect`
+    : undefined;
+
+  const bookmarkStorageKey = storageKey ? `${storageKey}-bookmarks` : undefined;
+  const systemNamespaceInSelectKey = storageKey
+    ? `${storageKey}-systemNamespaceInSelect`
+    : undefined;
+
+  const [bookmarks, setBookmarks] = useUserSettingsCompatibility(
+    bookmarkUserSettingsKey,
+    bookmarkStorageKey,
+    undefined,
+    true,
+  );
+
+  const [systemNamespacesInSelect, setSystemNamespacesInSelect] = useUserSettingsCompatibility(
+    systemNamespacesInSelectSettingsKey,
+    systemNamespaceInSelectKey,
+    false,
+    true,
+  );
+
+  const onBookmark = React.useCallback(
+    (key, active) => {
+      setBookmarks((oldBookmarks) => ({ ...oldBookmarks, [key]: active ? true : undefined }));
+    },
+    [setBookmarks],
+  );
+
+  const areSystemNamespaces = () => {
+    return options.some((option) => isSystemNamespace(option));
+  };
+
+  React.useEffect(() => {
+    if (isMainSelectOpen === true) {
+      setFilterText('');
+    }
+  }, [isMainSelectOpen]);
+
+  const onFavorite = (event: any, itemId: string) => {
+    const isCurrentBookmark = bookmarks && bookmarks[itemId] === true;
+    onBookmark(itemId, !isCurrentBookmark);
+  };
+
+  const showSystemSwitch = areSystemNamespaces() ? (
+    <>
+      <Divider />
+      <MenuGroup translate="true">
+        {/* NOTE: All Menu components throw type errors if translate isn't added */}
+        <MenuInput translate="true">
+          <Switch
+            id="no-label-switch-on"
+            label={
+              isProjects ? t('public~Show system projects') : t('public~Show system namespaces')
+            }
+            isChecked={systemNamespacesInSelect}
+            onChange={(isChecked: boolean) => {
+              setSystemNamespacesInSelect(isChecked);
+            }}
+            className="pf-c-select__menu-item pf-m-action co-namespace-selector__switch"
+          />
+        </MenuInput>
+      </MenuGroup>
+    </>
+  ) : null;
+
+  const filter = (
+    <MenuGroup translate="true">
+      <MenuInput translate="true">
+        <TextInput
+          autoFocus
+          value={filterText}
+          aria-label={isProjects ? t('public~Select project...') : t('public~Select namespace...')}
+          iconVariant="search"
+          type="search"
+          placeholder={isProjects ? t('public~Select project...') : t('public~Select namespace...')}
+          onChange={(value: string) => setFilterText(value)}
+          className="co-namespace-selector__filter"
+          ref={filterRef}
+        />
+      </MenuInput>
+    </MenuGroup>
+  );
+
+  const isFavorite = (option) => (bookmarks && bookmarks[option.key] ? true : false); // undefined cannot be an option
+
+  const isOptionShown = (option: { title: string; key: string }, checkIsFavorite?: boolean) => {
+    const containsFilterText = option.title.toLowerCase().includes(filterText.toLowerCase());
```

Comment:
@TheRealJon  This would change how the filtering works compared to the current project/namespace drop down.  In the current project/namespace drop down, the filter performs as an "include" meaning that matches from any position in the string will return true.  I am hesitant to implement the suggested change because only matches from the start of the string will be included.

I've used String.prototype.includes in numerous projects and haven't had a major issue so far.  Wondering if you can give me an example of where you all have encountered funky results so I can understand better.

Reply:
No need to change this if the existing behavior uses `includes`.